src/DocBlock/Tags/Template.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @link http://phpdoc.org
+ */
+
+namespace phpDocumentor\Reflection\DocBlock\Tags;
+
+use Doctrine\Deprecations\Deprecation;
+use phpDocumentor\Reflection\Type;
+use phpDocumentor\Reflection\DocBlock\Description;
+use phpDocumentor\Reflection\DocBlock\Tags\TagWithType;
+
+/**
+ * Reflection class for a {@}template tag in a Docblock.
+ */
+final class Template extends TagWithType 
+{
+	/** @var non-empty-string */
+	private $templateName;
+
+	/** @var ?Type */
+	private $bound;
+
+	/** @var ?Type */
+	private $default;
+
+    public function __construct(string $templateName, ?Type $bound = null, ?Type $default = null, ?Description $description = null)
+    {
+        $this->name         = 'template';
+        $this->templateName = $templateName;
+        $this->bound        = $bound;
```

Comment:
What is the bound property? I don't know what it means in this context. Can we clarify that by renaming the property or add some docblock that explains the value?

Reply:
I just copied property name from phpstan but as far I know:
`template T of int`
int is bound